libraries/AP_HAL_Linux/I2CDevice.cpp

```diff
@@ -177,13 +177,13 @@ bool I2CDevice::transfer(const uint8_t *send, uint32_t send_len,
     i2c_data.msgs = msgs;
     i2c_data.nmsgs = nmsgs;
 
-    int r = -EINVAL;
+    int r;
     unsigned retries = _retries;
     do {
         r = ::ioctl(_bus.fd, I2C_RDWR, &i2c_data);
-    } while (r < 0 && retries-- > 0);
+    } while (r == -1 && retries-- > 0);
```

Comment:
ioctl for i2c and spi devices in Linux return positive values for success - i.e. never a negative value. In the error condition a negative value (-1) is returned.

Is this patch so it matches the text in the man page?